app/controllers/locks_controller.rb

```diff
@@ -59,32 +58,30 @@ def format_errors(object)
     object.errors.messages.values.flatten.join("\n")
   end
 
-  def for_stage_lock?
-    case action_name
-    when 'create'
-      (params[:lock] || {})[:resource_type] == "Stage"
-    when 'destroy'
-      lock.resource_type == "Stage"
-    when 'index', 'destroy_via_resource'
-      false
-    else
-      raise 'Unsupported action'
-    end
+  def require_stage
+    @resource_class = "Stage"
+    @stage = find_resource
   end
 
-  def lock
-    @lock ||= Lock.find(params[:id])
+  def require_project
+    @resource_class = "Project"
+    @project = find_resource
   end
 
-  # Overrides CurrentStage#require_stage
-  def require_stage
+  def find_resource
     case action_name
     when 'create' then
-      @stage = Stage.find(params[:lock][:resource_id])
+      resource_class.find lock_params[:resource_id] if lock_params[:resource_type] == @resource_class
     when 'destroy' then
-      @stage = lock.resource
-    else
-      raise 'Unsupported action'
+      lock.resource if lock.resource_type == @resource_class
```

Comment:
we don't need this before-action on destroy ?

Reply:
we need this to determine, whether we need to assign the found resource to instance variable in require_stage and require_project methods.